### DISCUSSION TITLE
Re-using same policy definition multiple times in an initiative.

### DISCUSSION THREAD

**Discussion by dpb519** [gettek/terraform-azurerm-policy-as-code#97] [Ideas]
Hi Gettek,

There is a requirement within in our project to try to update the modules to have the ability to use same policy definition multiple times.

One way I am planning to achieve this is by checking for the reference id's and making sure there is no duplicate reference id.

variable.tf
```hcl
  all_reference_ids = flatten([
    for idx, d in var.member_definitions:
      ["${idx}", replace(substr(title(replace(d.name, "/-|_|\\s/", " ")), 0, 64), "/\\s/", "")]
 ])
```
main.tf

```hcl
   content {
      policy_definition_id = policy_definition_reference.value.id
      #reference_id         = policy_definition_reference.value.ref_id
      reference_id = index(local.all_reference_ids, policy_definition_reference.value.ref_id) > 1 ? "${policy_definition_reference.value.ref_id}${count.index +1}" : policy_definition_reference.value.ref_id
      parameter_values = length(policy_definition_reference.value.parameters) > 0 ? jsonencode({
        for k in keys(policy_definition_reference.value.parameters) :
        k => {
          value = k == "effect" && var.merge_effects == false ? "[parameters('${format("%s_%s", k, policy_definition_reference.value.ref_id)}')]" : var.merge_parameters == false ? "[parameters('${format("%s_%s", k, policy_definition_reference.value.ref_id)}')]" : "[parameters('${k}')]"
        }
      }) : null
      policy_group_names = policy_definition_reference.value.groups
    }
```
however I am having challenges in creating the parameters and roledefinition ids. 

So wanted to check with you if this approach will work. 

PS. I have already tried #67 . however we wanted this to be achieved by updating the modules.

**Comment by gettek** [comment]
Hi @dpb519,

This has been a long standing feature request and I've not been able to give it the attention it needs unfortunately.

I like your approach using the loop `index()` but as you mentioned building the initiative parameters is the tricky part. If you could provide a baseline solution, I'll try and devote some equal time into getting this implemented. Thanks!

**Reply by gettek** [reply]
@dpb519 please see the latest changes released in v2.9.0 (#100)
